src/apps/js_generic/js_generic.cpp

```diff
@@ -268,6 +280,42 @@ namespace ccfapp
     const auto property_name = JS_AtomToCString(ctx, property);
     LOG_TRACE_FMT("Looking for kv map '{}'", property_name);
 
+    const auto [security_domain, access_category] =
+      kv::parse_map_name(property_name);
+
+    auto read_only = false;
+    switch (access_category)
+    {
+      case kv::AccessCategory::INTERNAL:
+      {
+        if (security_domain == kv::SecurityDomain::PUBLIC)
+        {
+          read_only = true;
+        }
+        else
+        {
+          throw std::runtime_error(fmt::format(
+            "JS application cannot access private internal CCF table '{}'",
```

Comment:
I think "Application" may be more accurate than "JS Application", although it is the JS App here, the reason it's denied access if that it's app code, whether it's JS or not isn't relevant.